File: uikit-sample/src/main/java/com/sendbird/uikit_messaging_android/utils/PreferenceUtils.java
package com.sendbird.uikit_messaging_android.utils;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;

import androidx.annotation.NonNull;

/**
 * This provides methods to manage preferences data.
 */
public class PreferenceUtils {

    private static final String PREFERENCE_KEY_USER_ID = "PREFERENCE_KEY_USER_ID";
    private static final String PREFERENCE_KEY_NICKNAME = "PREFERENCE_KEY_NICKNAME";
    private static final String PREFERENCE_KEY_PROFILE_URL = "PREFERENCE_KEY_PROFILE_URL";
    private static final String PREFERENCE_KEY_USE_DARK_THEME = "PREFERENCE_KEY_USE_DARK_THEME";
    private static final String PREFERENCE_KEY_DO_NOT_DISTURB = "PREFERENCE_KEY_DO_NOT_DISTURB";

    @SuppressLint("StaticFieldLeak")
    private static Context context;

    // Prevent instantiation
    private PreferenceUtils() {
    }

    public static void init(@NonNull Context appContext) {
        context = appContext.getApplicationContext();
    }

    private static SharedPreferences getSharedPreferences() {
        return context.getSharedPreferences("sendbird", Context.MODE_PRIVATE);
    }

    public static void setUserId(@NonNull String userId) {
        SharedPreferences.Editor editor = getSharedPreferences().edit();
        editor.putString(PREFERENCE_KEY_USER_ID, userId).apply();
    }

    @NonNull
    public static String getUserId() {
        final String value = getSharedPreferences().getString(PREFERENCE_KEY_USER_ID, "");
        return value == null ? "" : value;
    }

    public static void setNickname(@NonNull String nickname) {
        SharedPreferences.Editor editor = getSharedPreferences().edit();
        editor.putString(PREFERENCE_KEY_NICKNAME, nickname).apply();
    }

    @NonNull
    public static String getNickname() {
        final String value = getSharedPreferences().getString(PREFERENCE_KEY_NICKNAME, "");
        return value == null ? "" : value;
    }

    public static void setProfileUrl(@NonNull String profileUrl) {
        SharedPreferences.Editor editor = getSharedPreferences().edit();
        editor.putString(PREFERENCE_KEY_PROFILE_URL, profileUrl).apply();
    }

    @NonNull
    public static String getProfileUrl() {
        final String value = getSharedPreferences().getString(PREFERENCE_KEY_PROFILE_URL, "");
        return value == null ? "" : value;
    }

    public static void setUseDarkTheme(boolean useDarkTheme) {
        SharedPreferences.Editor editor = getSharedPreferences().edit();
        editor.putBoolean(PREFERENCE_KEY_USE_DARK_THEME, useDarkTheme).apply();
    }

    public static boolean isUsingDarkTheme() {
        return getSharedPreferences().getBoolean(PREFERENCE_KEY_USE_DARK_THEME, false);
    }

    public static void setDoNotDisturb(boolean doNotDisturb) {
        SharedPreferences.Editor editor = getSharedPreferences().edit();
        editor.putBoolean(PREFERENCE_KEY_DO_NOT_DISTURB, doNotDisturb).apply();
    }

    public static boolean getDoNotDisturb() {
        return getSharedPreferences().getBoolean(PREFERENCE_KEY_DO_NOT_DISTURB, false);
    }

    public static void clearAll() {
        SharedPreferences.Editor editor = getSharedPreferences().edit();
        editor.clear().apply();
    }
}

File: uikit-sample/src/main/kotlin/com/sendbird/uikit_messaging_android/utils/PreferenceUtils.kt
package com.sendbird.uikit_messaging_android.utils

import android.annotation.SuppressLint
import android.content.Context
import android.content.SharedPreferences

/**
 * This provides methods to manage preferences data.
 */
@SuppressLint("StaticFieldLeak")
object PreferenceUtils {
    private const val PREFERENCE_KEY_APP_ID = "PREFERENCE_KEY_APP_ID"
    private const val PREFERENCE_KEY_USER_ID = "PREFERENCE_KEY_USER_ID"
    private const val PREFERENCE_KEY_NICKNAME = "PREFERENCE_KEY_NICKNAME"
    private const val PREFERENCE_KEY_PROFILE_URL = "PREFERENCE_KEY_PROFILE_URL"
    private const val PREFERENCE_KEY_USE_DARK_THEME = "PREFERENCE_KEY_USE_DARK_THEME"
    private const val PREFERENCE_KEY_DO_NOT_DISTURB = "PREFERENCE_KEY_DO_NOT_DISTURB"

    private lateinit var context: Context
    fun init(appContext: Context) {
        context = appContext.applicationContext
    }

    private val sharedPreferences: SharedPreferences
        get() = context.getSharedPreferences("sendbird", Context.MODE_PRIVATE)
    var appId: String
        get() = sharedPreferences.getString(PREFERENCE_KEY_APP_ID, "") ?: ""
        @SuppressLint("ApplySharedPref")
        set(appId) {
            val editor = sharedPreferences.edit()
            editor.putString(PREFERENCE_KEY_APP_ID, appId).commit()
        }

    var userId: String
        get() {
            val value = sharedPreferences.getString(PREFERENCE_KEY_USER_ID, "")
            return value ?: ""
        }
        set(userId) {
            val editor = sharedPreferences.edit()
            editor.putString(PREFERENCE_KEY_USER_ID, userId).apply()
        }

    var nickname: String
        get() {
            val value = sharedPreferences.getString(PREFERENCE_KEY_NICKNAME, "")
            return value ?: ""
        }
        set(nickname) {
            val editor = sharedPreferences.edit()
            editor.putString(PREFERENCE_KEY_NICKNAME, nickname).apply()
        }

    var profileUrl: String
        get() {
            val value = sharedPreferences.getString(PREFERENCE_KEY_PROFILE_URL, "")
            return value ?: ""
        }
        set(profileUrl) {
            val editor = sharedPreferences.edit()
            editor.putString(PREFERENCE_KEY_PROFILE_URL, profileUrl).apply()
        }

    var isUsingDarkTheme: Boolean
        get() = sharedPreferences.getBoolean(PREFERENCE_KEY_USE_DARK_THEME, false)
        set(useDarkTheme) {
            val editor = sharedPreferences.edit()
            editor.putBoolean(PREFERENCE_KEY_USE_DARK_THEME, useDarkTheme).apply()
        }

    var doNotDisturb: Boolean
        get() = sharedPreferences.getBoolean(PREFERENCE_KEY_DO_NOT_DISTURB, false)
        set(doNotDisturb) {
            val editor = sharedPreferences.edit()
            editor.putBoolean(PREFERENCE_KEY_DO_NOT_DISTURB, doNotDisturb).apply()
        }

    fun clearAll() {
        val editor = sharedPreferences.edit()
        editor.clear().apply()
    }
}
